plugins/dates/src/api/parse/range/02-date-range.js

```diff
@@ -138,4 +139,30 @@ export default [
       return null
     },
   },
+
+  {
+    // 2 to 4 weeks
+    match: '[<min>#Value] to [<max>#Value] [<unit>(day|days|week|weeks|month|months|year|years)]',
+    desc: '2 to 4 weeks',
+    parse: (m, context) => {
+      const { min, max, unit } = m.groups()
+
+      let start = new Unit(context.today, null, context)
+      let end = start.clone()
+
+      const duration = unit.text('implicit')
+      start = start.applyShift({ [duration]: min.numbers().get()[0] })
+      end = end.applyShift({ [duration]: max.numbers().get()[0] })
+
+      // Ensure that the end date is inclusive
+      if (!['day', 'days'].includes(duration)) {
+        end = end.applyShift({ day: -1 }).applyShift({ [duration]: 1 })
+      }
```

Comment:
Hopefully this is an okay way of doing this.
I found that day and days was projecting inclusively, but, other durations weren't.

For example `1-2 weeks` was returned as 8 days rather than 14 days.

Reply:
sounds good!

Reply:
Great, thanks @spencermountain!
Hopefully last question, do i need to do anything extra for handling past tense in this?
I.e. `2 to 3 weeks ago`.

Reply:
oh, no problem - i would make a new section, in addition to this '2 to 4 weeks' one, with similar logic.
go for it!

oh,  one thing - maybe add a `'^in'` on the match, so it doesn't interfere with `2 to 3 weeks after june 3rd` etc.
lookin good! thanks for the help.

Reply:
Cheers @spencermountain, have added the `^in` as requested for the future tense match block.

Struggling with the past tense, as you can see from the now failing tests! When you get a spare second could you take a look at the past test block please? Thought it'd be as simple as flipping the logic.

Reply:
sure - i can merge this to dev and fix the failing tests, if you're ready?

Reply:
hey Frazer, haven't forgotten about this -  been really busy, then had covid. Will try to get to it shortly.
cheers

Reply:
No worries!

Reply:
Thanks for taking this on @spencermountain, https://github.com/spencermountain/compromise/commit/42f9cd0b6ee40401ad89fb375fd2207de3757a5d appears to have stopped the ranges from being inclusive?